File: chain-sdk/javascript-closure-angular/API/Client/TickV3DTO.js
goog.provide('API.Client.TickV3DTO');

/**
 * Ticks are the boundaries between discrete areas in price space.
 * @record
 */
API.Client.TickV3DTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.TickV3DTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.TickV3DTO.prototype.recvTime;

/**
 * Number of block in which entity was recorded.
 * @type {!number}
 * @export
 */
API.Client.TickV3DTO.prototype.blockNumber;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.TickV3DTO.prototype.vid;

/**
 * Identifier, format: <pool address>#<tick index>
 * @type {!string}
 * @export
 */
API.Client.TickV3DTO.prototype.id;

/**
 * Pool address.
 * @type {!string}
 * @export
 */
API.Client.TickV3DTO.prototype.poolAddress;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.TickV3DTO.prototype.tickIdx;

/**
 * Pool address.
 * @type {!string}
 * @export
 */
API.Client.TickV3DTO.prototype.pool;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.TickV3DTO.prototype.liquidityGross;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.TickV3DTO.prototype.liquidityNet;

/**
 * Calculated price of token0 of tick within this pool - constant.
 * @type {!string}
 * @export
 */
API.Client.TickV3DTO.prototype.price0;

/**
 * Calculated price of token1 of tick within this pool - constant.
 * @type {!string}
 * @export
 */
API.Client.TickV3DTO.prototype.price1;

/**
 * Lifetime volume of token0 with this tick in range.
 * @type {!string}
 * @export
 */
API.Client.TickV3DTO.prototype.volumeToken0;

/**
 * Lifetime volume of token1 with this tick in range.
 * @type {!string}
 * @export
 */
API.Client.TickV3DTO.prototype.volumeToken1;

/**
 * Lifetime volume in derived USD with this tick in range.
 * @type {!string}
 * @export
 */
API.Client.TickV3DTO.prototype.volumeUsd;

/**
 * Lifetime volume in untracked USD with this tick in range.
 * @type {!string}
 * @export
 */
API.Client.TickV3DTO.prototype.untrackedVolumeUsd;

/**
 * Fees in USD.
 * @type {!string}
 * @export
 */
API.Client.TickV3DTO.prototype.feesUsd;

/**
 * All time collected fees in token0.
 * @type {!string}
 * @export
 */
API.Client.TickV3DTO.prototype.collectedFeesToken0;

/**
 * All time collected fees in token1.
 * @type {!string}
 * @export
 */
API.Client.TickV3DTO.prototype.collectedFeesToken1;

/**
 * All time collected fees in USD.
 * @type {!string}
 * @export
 */
API.Client.TickV3DTO.prototype.collectedFeesUsd;

/**
 * Created time.
 * @type {!Date}
 * @export
 */
API.Client.TickV3DTO.prototype.createdAtTimestamp;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.TickV3DTO.prototype.liquidityProviderCount;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.TickV3DTO.prototype.feeGrowthOutside0x128;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.TickV3DTO.prototype.feeGrowthOutside1x128;

File: chain-sdk/clojure/src/on_chain_api/specs/tick_v3_dto.clj
(ns on-chain-api.specs.tick-v3-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [on-chain-api.specs.big-integer :refer :all]
            [on-chain-api.specs.big-integer :refer :all]
            [on-chain-api.specs.big-integer :refer :all]
            [on-chain-api.specs.big-integer :refer :all]
            [on-chain-api.specs.big-integer :refer :all]
            [on-chain-api.specs.big-integer :refer :all]
            )
  (:import (java.io File)))


(def tick-v3-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :vid) int?
   (ds/opt :id) string?
   (ds/opt :pool_address) string?
   (ds/opt :tick_idx) big-integer-spec
   (ds/opt :pool) string?
   (ds/opt :liquidity_gross) big-integer-spec
   (ds/opt :liquidity_net) big-integer-spec
   (ds/opt :price_0) string?
   (ds/opt :price_1) string?
   (ds/opt :volume_token_0) string?
   (ds/opt :volume_token_1) string?
   (ds/opt :volume_usd) string?
   (ds/opt :untracked_volume_usd) string?
   (ds/opt :fees_usd) string?
   (ds/opt :collected_fees_token_0) string?
   (ds/opt :collected_fees_token_1) string?
   (ds/opt :collected_fees_usd) string?
   (ds/opt :created_at_timestamp) inst?
   (ds/opt :liquidity_provider_count) big-integer-spec
   (ds/opt :fee_growth_outside_0x128) big-integer-spec
   (ds/opt :fee_growth_outside_1x128) big-integer-spec
   })

(def tick-v3-dto-spec
  (ds/spec
    {:name ::tick-v3-dto
     :spec tick-v3-dto-data}))
